app/controllers/api/v1/orders_controller.rb

```diff
@@ -73,6 +73,16 @@ def transition
         render json: serializer.new(@order, include_allowed_actions: true)
       end
 
+      def fetch_shipment_or_carryout_code
+        record = Order.get_subsequent_international_code(params['detail_type'])
+
+        if record <= Order::Type::MAX_INTERNATIONAL_ORDER_CODE
+          render json: { code: record }
+        else
+          render json: { errors: "Code Limit Exhausted" }, status: 404
+        end
```

Comment:
No need to do this check here. You can rewrite the method somewhat like this

```suggestion
        code = Order.get_next_code_for(params['detail_type'])

        render json: { code: record }
```
render error in case there is an exception or if code is nil/invalid